### PR TITLE
52 fix weak password error message

### PR DESCRIPTION
**Pull request by NDenheim**

### CODE REVIEW

File: src/pages/Register/Register.tsx [hunk 1]
```diff
@@ -42,7 +42,11 @@ const Register = ({ setUser }: RegisterProps) => {
       navigate("/");
     } catch (error: unknown) {
       if (error instanceof FirebaseError) {
-        if (error.code === "auth/email-already-in-use") setIsEmailExist(true);
+        if (error.code === "auth/email-already-in-use")
+          setFirebaseError("emailError");
```
Comment (Charlie-robin) [inline src/pages/Register/Register.tsx:46]: I think you can just add the text you want to display to state.

```jsx
if (error.code === "auth/email-already-in-use")
          setFirebaseError("Email Already Exists");
 }         
 
 // Then in return, you only need one check
 
{firebaseError && <p className="register-page__email-exists">
{firebaseError }</p>}
```

File: src/pages/Register/Register.tsx [hunk 2]
```diff
@@ -195,6 +193,23 @@ const Register = ({ setUser }: RegisterProps) => {
                 handleInput={handlePasswordInput}
               />
             </div>
+            <div>
+              {firebaseError === "emailError" && (
+                <>
```
Comment (Charlie-robin) [inline src/pages/Register/Register.tsx:198]: You do not need a react fragment `<>` as you are displaying one p tag, not multiples.

**Review by Charlie-robin — CHANGES_REQUESTED** [review]
Some small tweaks please

File: src/pages/Register/Register.tsx (hunk 1 above)
Comment (NDenheim) [reply src/pages/Register/Register.tsx]: Done!

**Review by Charlie-robin — APPROVED** [review]
Looks Great 👍

File: src/pages/Register/Register.tsx [hunk 3]
```diff
@@ -195,6 +193,15 @@ const Register = ({ setUser }: RegisterProps) => {
                 handleInput={handlePasswordInput}
               />
             </div>
+            <div>
+              {firebaseError && (
+                <p className="register-page__email-exists">{firebaseError}</p>
+              )}
+              {firebaseError === "passwordError" && (
+                <p className="register-page__weak-password">{firebaseError}</p>
+              )}
```
Comment (olirob93) [inline src/pages/Register/Register.tsx:202]: remove this it's not needed

File: src/pages/Register/Register.scss [hunk 4]
```diff
@@ -83,9 +83,12 @@
...
-    left: 0;
     width: 100%;
-    z-index: 1;
+  }
+
+  &__weak-password {
+    color: $color-red;
+    text-align: center;
+    width: 100%;
   }
```
Comment (olirob93) [inline src/pages/Register/Register.scss:93]: remove this not needed

File: src/pages/Register/Register.tsx [hunk 5]
```diff
@@ -195,6 +193,15 @@ const Register = ({ setUser }: RegisterProps) => {
                 handleInput={handlePasswordInput}
               />
             </div>
+            <div>
+              {firebaseError && (
+                <p className="register-page__email-exists">{firebaseError}</p>
```
Comment (olirob93) [inline src/pages/Register/Register.tsx:198]: change classname to register-page__firebase-error

File: src/pages/Register/Register.scss [hunk 6]
```diff
@@ -83,9 +83,12 @@
   &__email-exists {
```
Comment (olirob93) [inline src/pages/Register/Register.scss:83]: change this to match the comment below __firebase-error

**Review by olirob93 — CHANGES_REQUESTED** [review]

**Review by olirob93 — APPROVED** [review]

**Pull request merged**